photon-lib/src/main/scala/com/linkedin/photon/ml/data/ScoredGameDatum.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.data
+
+import scala.collection.Map
+
+import com.linkedin.photon.ml.util.MathUtils.isAlmostZero
+
+/**
+ * Representation of a single scored GAME data point. This data structure handles the transition from a training
+ * example with features to an example that has been scored. Adding a score field to GameDatum is infeasible because
+ * after scoring, the feature shards are no longer needed and keeping them around increases the cost to shuffle the
+ * data structure around
+ *
+ * @param response The response or label
+ * @param offset offset
+ * @param weight importance weight
+ * @param score computed score for this instance
+ * @param idTypeToValueMap The id type to value map that holds different types of ids associated with this data
```

Comment:
Since `ScoredGameDatum` is a case class, it has an `apply` method auto-generated. Alternatively, we could have it not be a case class and add another `apply` method with default values, but this would have identical behaviour unless we made the constructor private/protected, in which case the apply method is just forcing the code to use an identical apply method instead of the constructor